### PR TITLE
refactor: Change focus and validation behavior of password inputs

### LINKED ISSUE #222: refactor: Trigger confirm password validation after losing focus
### Description

When confirming a password during profile creation, the validation triggers immediately after typing the first character. 

The validation should trigger only when the field loses focus.

Spin-off issue from https://github.com/impierce/identity-wallet/issues/211.

### Motivation

Create a polished UX.

### Resources

_No response_

### To-do List

- [ ] Change the validation behavior.

### PR DESCRIPTION
**Pull request by maiertech**
# Description of change

Route `/welcome/password`:

1. The password input element gets initial focus.
2. The password input element gets the focus back when toggling password visibility.

Route `/welcome/password/confirm`:

1. The password input element gets initial focus.
2. The validation triggers only when the passwords match or when the input element loses focus.
3. The password input element gets the focus back when toggling password visibility.

## Links to any relevant issues

Fixes #222.

## How the change has been tested

Manual visual testing. Reset the app in dev mode and walk through the onboarding process. Toggle visibility and make the input element lose focus.

## Definition of Done checklist

Add an `x` to the boxes that are relevant to your changes.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: unime/src/routes/welcome/password/+page.svelte [hunk 1]
```diff
@@ -16,10 +18,18 @@
 
   let passwordPolicyViolations: string[] = checkPasswordPolicy($onboarding_state.password ?? '');
   let showPassword = false;
+
+  // Ref to input DOM element.
+  let input_element: HTMLInputElement;
+
+  onMount(() => {
+    // Initial focus.
```
Comment (daniel-mader) [inline unime/src/routes/welcome/password/+page.svelte]: I don't think we need this comment as the code explains itself.

File: unime/src/routes/welcome/password/confirm/+page.svelte [hunk 2]
```diff
@@ -12,12 +14,23 @@
   import Smiley from '~icons/ph/smiley';
   import SmileySad from '~icons/ph/smiley-sad';
 
-  let passwordsEqual: boolean | undefined;
+  // 3 states: true (match), false (mismatch), undefined (I don't know).
```
Comment (daniel-mader) [inline unime/src/routes/welcome/password/confirm/+page.svelte]: I would update `"I don't know"` with `"unknown"` or `"not checked yet"`.

File: unime/src/routes/welcome/password/confirm/+page.svelte [hunk 3]
```diff
@@ -12,12 +14,23 @@
...
+  let passwords_match: boolean | undefined = undefined;
```
Comment (daniel-mader) [inline unime/src/routes/welcome/password/confirm/+page.svelte]: Let's stick with the official naming convention where possible (camelCase). We're only using snake_case for types exported by Rust, but we should clean that up eventually too.

File: unime/src/routes/welcome/password/confirm/+page.svelte [hunk 4]
```diff
@@ -12,12 +14,23 @@
...
+
   let showPassword = false;
+  let password = '';
+
+  // Ref to input DOM element.
+  let input_element: HTMLInputElement;
+
+  onMount(() => {
+    // Initial focus.
```
Comment (daniel-mader) [inline unime/src/routes/welcome/password/confirm/+page.svelte]: I don't think we need this comment as the code explains itself.

File: unime/src/routes/welcome/password/confirm/+page.svelte [hunk 5]
```diff
@@ -29,14 +42,39 @@
...
+        }
+      }}
+      class="h-12 w-full rounded-xl border border-slate-300 bg-white px-4 py-3 text-[13px]/[24px] text-slate-500 dark:border-slate-600 dark:bg-dark dark:text-slate-300"
     />
     <div class="absolute right-3 top-0 flex h-full items-center">
-      <button class="rounded-full p-2" on:click={() => (showPassword = !showPassword)}>
+      <button
+        class="rounded-full p-2"
+        on:click={() => {
+          // Focus input element when toggling visbility.
```
Comment (daniel-mader) [inline unime/src/routes/welcome/password/confirm/+page.svelte]: Typo: `visbility` --> `visibility`

**Review by daniel-mader — CHANGES_REQUESTED** [review]

File: unime/src/routes/welcome/password/+page.svelte (hunk 1 above)
Comment (maiertech) [reply unime/src/routes/welcome/password/+page.svelte]: Comment removed.

File: unime/src/routes/welcome/password/confirm/+page.svelte (hunk 5 above)
Comment (maiertech) [reply unime/src/routes/welcome/password/confirm/+page.svelte]: Typo fixed.

File: unime/src/routes/welcome/password/confirm/+page.svelte (hunk 4 above)
Comment (maiertech) [reply unime/src/routes/welcome/password/confirm/+page.svelte]: Comment removed.

File: unime/src/routes/welcome/password/confirm/+page.svelte (hunk 3 above)
Comment (maiertech) [reply unime/src/routes/welcome/password/confirm/+page.svelte]: Was not aware of this convention.

File: unime/src/routes/welcome/password/confirm/+page.svelte (hunk 2 above)
Comment (maiertech) [reply unime/src/routes/welcome/password/confirm/+page.svelte]: Comment changed.

**Comment by SoulaAndrikopoulos** [comment]
From a usability perspective: As we discussed earlier it would be nice to still have an indicator when the passwords don't match, just not immediately at the beginning when the user starts typing :) Otherwise not a lot to add to the other comments.

**Review by daniel-mader — APPROVED** [review]

File: unime/src/routes/welcome/password/confirm/+page.svelte [hunk 6]
```diff
@@ -45,9 +93,9 @@
       </button>
     </div>
   </div>
-  {#if passwordsEqual !== undefined}
+  {#if passwordsMatch !== undefined}
```
Comment (SoulaAndrikopoulos) [inline unime/src/routes/welcome/password/confirm/+page.svelte:96]: This naming is easier to understand 👍

**Review by SoulaAndrikopoulos — APPROVED** [review]
Code is more clear and comments have been added, great work. The validation behavior has been tested and works fine in the app.

**Pull request merged**